table/get_context.cc

```diff
@@ -561,17 +566,31 @@ void replayGetContextLog(const Slice& replay_log, const Slice& user_key,
                          GetContext* get_context, Cleanable* value_pinner,
                          SequenceNumber seq_no) {
   Slice s = replay_log;
+  Slice ts = Slice();
+  size_t ts_sz = get_context->TimestampSize();
+  bool ret;
+
   while (s.size()) {
     auto type = static_cast<ValueType>(*s.data());
     s.remove_prefix(1);
     Slice value;
-    bool ret = GetLengthPrefixedSlice(&s, &value);
+    ret = GetLengthPrefixedSlice(&s, &value);
     assert(ret);
     (void)ret;
 
     bool dont_care __attribute__((__unused__));
 
     ParsedInternalKey ikey = ParsedInternalKey(user_key, seq_no, type);
+
+    // If ts enabled for current cf, there will always be ts appended after each
+    // piece of value.
+    if (ts_sz > 0) {
+      ret = GetLengthPrefixedSlice(&s, &ts);
+      assert(ret);
```

Comment:
nit: it would be good to also assert with `assert(ts.size() == ts_sz);` here so we have it on both the read and write paths.